File: app/src/main/java/com/kksionek/gdzietentramwaj/di/AppComponent.java
package com.kksionek.gdzietentramwaj.di;

import android.content.Context;

import com.kksionek.gdzietentramwaj.Repository.LocationRepository;
import com.kksionek.gdzietentramwaj.Repository.TramRepository;
import com.kksionek.gdzietentramwaj.ViewModel.MainActivityViewModel;

import javax.inject.Singleton;

import dagger.Component;

@Singleton
@Component(modules = {AppModule.class})
public interface AppComponent {
    TramRepository getTramRepository();
    LocationRepository getLocationRepository();
    Context getAppContext();
}

File: app/src/main/java/com/kksionek/gdzietentramwaj/di/AppComponent.kt
package com.kksionek.gdzietentramwaj.di

import com.kksionek.gdzietentramwaj.view.FavoriteLinesActivity
import com.kksionek.gdzietentramwaj.view.MapsActivity
import dagger.Component
import javax.inject.Singleton

@ActivityScope
@Singleton
@Component(
    modules = [
        AppModule::class,
        ViewModelFactoryModule::class,
        MapsActivityModule::class,
        FavoriteLinesActivityViewModelModule::class
    ]
)
interface AppComponent {
    fun inject(activity: MapsActivity)
    fun inject(activity: FavoriteLinesActivity)
}
